flake.nix

```diff
@@ -15,14 +27,183 @@
+                             src_files+=`find ./tls -name .git -prune -o -regextype posix-egrep -regex "$include_regex" -print`
+                             src_files+=" "
+                             src_files+=`find ./utils -name .git -prune -o -regextype posix-egrep -regex "$include_regex" -print`
+                             src_files+=" "
+                             src_files+=`find ./tests/unit -name .git -prune -o -regextype posix-egrep -regex "$include_regex" -print`
+                             src_files+=" "
+                             src_files+=`find ./tests/testlib -name .git -prune -o -regextype posix-egrep -regex "$include_regex" -print`
+            		             echo $src_files | xargs -n 1 -P $(nproc) clang-format --dry-run -style=file                 
+                           }
+                          function do-clang-format {
+                             banner "In place clang-format"
+                             cd $SRC_ROOT
+                             include_regex=".*\.(c|h)$"
+                             src_files=`find ./api -name .git -prune -o -regextype posix-egrep -regex "$include_regex" -print`
+                             src_files+=" "
+                             src_files+=`find ./bin -name .git -prune -o -regextype posix-egrep -regex "$include_regex" -print`
+                             src_files+=" "
+                             src_files+=`find ./crypto -name .git -prune -o -regextype posix-egrep -regex "$include_regex" -print`
+                             src_files+=" "
+                             src_files+=`find ./stuffer -name .git -prune -o -regextype posix-egrep -regex "$include_regex" -print`
+                             src_files+=" "
+                             src_files+=`find ./error -name .git -prune -o -regextype posix-egrep -regex "$include_regex" -print`
+                             src_files+=" "
+                             src_files+=`find ./tls -name .git -prune -o -regextype posix-egrep -regex "$include_regex" -print`
+                             src_files+=" "
+                             src_files+=`find ./utils -name .git -prune -o -regextype posix-egrep -regex "$include_regex" -print`
+                             src_files+=" "
+                             src_files+=`find ./tests/unit -name .git -prune -o -regextype posix-egrep -regex "$include_regex" -print`
+                             src_files+=" "
+                             src_files+=`find ./tests/testlib -name .git -prune -o -regextype posix-egrep -regex "$include_regex" -print`
+            		             echo $src_files | xargs -n 1 -P $(nproc) clang-format -style=file -i
+                           }
+          '';
+
+          packages = [
+            # Build Depends
+            openssl_1_1_1
+            pkgs.cmake
+            # Other Libcryptos
+            # openssl_0_9_8 openssl_1_0_2 openssl_3_0
+            # libressl pkgs.boringssl
```

Comment:
Is it possible to have multiple dev shells where you can select these?

Reply:
We do that for build, but it doesn't make as much sense here. Often if you are debugging something with openssl you'll want access to all of them. I do think that it is a little silly to have them all commented out here though. Instead I propose putting them all in the environment and creating alias so you can run all the openssls.